crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -834,6 +735,147 @@ impl ScheduleGraph {
         }
     }
 
+    /// Adds the config nodes to the graph.
+    ///
+    /// `process_config` is the function which processes each config node and returns a corresponding `NodeId`.
+    ///
+    /// The fields on the returned [`ProcessConfigsResult`] are:
+    /// - `nodes`: a vector of all node ids contained in the nested `NodeConfigs`
+    /// - `densely_chained`: a boolean that is true if all nested nodes are linearly chained (with successive `after` orderings) in the order they are defined
+    #[track_caller]
+    fn process_configs<T: ProcessNodeConfig>(
+        &mut self,
+        configs: NodeConfigs<T>,
+    ) -> ProcessConfigsResult {
+        match configs {
+            NodeConfigs::NodeConfig(_) => ProcessConfigsResult {
+                densely_chained: true,
+                nodes: Vec::new(),
+            },
+            NodeConfigs::Configs {
+                mut configs,
+                collective_conditions,
+                chained,
+            } => {
+                let more_than_one_entry = configs.len() > 1;
+                configs = self.apply_config_conditions(configs, collective_conditions);
+                let mut config_iter = configs.into_iter();
+                let mut densely_chained = true;
+                if chained == Chain::Yes || chained == Chain::YesIgnoreDeferred {
+                    let Some(prev) = config_iter.next() else {
+                        return ProcessConfigsResult {
+                            nodes: Vec::new(),
+                            densely_chained: true,
+                        };
+                    };
+                    let mut previous_result = self.process_configs_and_collect_nodes(prev);
+                    densely_chained = previous_result.densely_chained;
+                    for current in config_iter {
+                        let current_result = self.process_configs_and_collect_nodes(current);
+                        densely_chained = densely_chained && current_result.densely_chained;
+                        self.chain(chained, &mut previous_result, &current_result);
+
+                        previous_result = current_result;
+                    }
+                } else {
+                    for config in config_iter {
+                        let result = self.process_configs(config);
+                        densely_chained = densely_chained && result.densely_chained;
+                    }
+
+                    // an "unchained" SystemConfig is only densely chained if it has exactly one densely chained entry
+                    if more_than_one_entry {
+                        densely_chained = false;
+                    }
+                }
+
+                ProcessConfigsResult {
+                    nodes: Vec::new(),
+                    densely_chained,
+                }
+            }
+        }
+    }
+
+    fn chain(
```

Comment:
Doc strings would be nice here.

Reply:
Are doc strings the same as doc comments? I was not sure exactly what the extracted functions do, I only guessed the names based on the context, thus I do not know what to write as doc comments.